profile/meson.build

```diff
@@ -6,5 +6,5 @@ libass_profile = executable(
     install: false,
     include_directories: incs,
     dependencies: deps,
-    link_with: libass_for_tools,
+    objects: libass.extract_all_objects(recursive: true),
```

Comment:
Only checkasm and compare need a static lib, the rest can work with either (thus they can also be used to test dynamic linking works)

Reply:
In this case it should link to libass directly.

Reply:
No it shouldn’t. Linking to libass directly prefers dynamic, while we want to prefer but not require static

Reply:
Why? As I understand the libass_for_tools were added only to make those tests work in 'both' build.

Why do you want to prefer static? I'm asking because it has to be documented.

Reply:
> As I understand the libass_for_tools were added only to make those tests work in 'both' build.

It wasn’t and if it were, would only have been added to the relevant utilities. `libass_for_tools` exists since the original meson implementation predating any tests being hooked up. No changes or comments wrt to `libass_for_tools`are needed.
 
- Statically linked, uninstalled binaries are resilient to directory renaming, can be copied around and e.g. kept as a reference during bisects
- preferring static matches the autotools builds *(there a simple libtool link flag suffices to prefer static when both are available)*

Reply:
I understand why static tools may be preferable. Sorry, it's my fault for not asking questions clearly. If you prefer statically linked tools, why don't you want that when building shared libass?

> preferring static matches the autotools builds (there a simple libtool link flag suffices to prefer static when both are available)

Meson has `prefer_static` flag. I don't see how autotools are relevant here?

Reply:
Utilities are never installed and serve as convenient tools for developers and testing; therefore it makes sense to tweak the process for it and since it’ſ not installed we can be more opinionated.
The libass library on the other hand is the main part installed on systems later, here it should follow whatever the builder chose. Following/Keeping common defaults unless there’s a reason not too (like with the default library type) is also good to yield expected results.

I believe `prefer_static` used to require a too-new meson version for most of the PRs lifetime. But even now, `prefer_static` affects _all_ libraries iiuc, while we mostly just care about our own code during e.g. bisects.

Consistency with autotools matters insofar as it means there are less differences to keep in mind when switching back and forth between the build systems during testing creating less surprises. *(Or if you meant why I mentioned how the behaviour is achieved in autotools: so there’s no need to search for a non-existing `libass_for_tools` equivalent as a reference in the autotools build setup)*